### PR TITLE
Mtd 10 service

### PR DESCRIPTION
**Pull request by ghilyana**

### CODE REVIEW

File: src/main/java/com/ghilly/repository/CountryRepository.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+public interface CountryRepository {
+    void insert(String countryName);
+
+    List<String> takeAllCountries();
+
+    String takeCountry(int countryId);
+
+    void change(int countryId, String newName);
+
+    void cut(int countryId);
```
Comment (gregorusanov) [inline src/main/java/com/ghilly/repository/CountryRepository.java]: please rename to remove or delete

File: src/main/java/com/ghilly/repository/CountryRepository.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.List;
+
+public interface CountryRepository {
+    void insert(String countryName);
+
+    List<String> takeAllCountries();
+
+    String takeCountry(int countryId);
+
+    void change(int countryId, String newName);
```
Comment (gregorusanov) [inline src/main/java/com/ghilly/repository/CountryRepository.java]: please rename to update or upgrade

File: src/main/java/com/ghilly/service/CountryServiceRest.java [hunk 3]
```diff
@@ -1,43 +1,58 @@
...
+        logger.info("The list of countries is:");
+        return repository.takeAllCountries();
     }
 
     @Override
     public String getCountry(int countryId) {
-        logger.info("getCountry");
-        return "There will be the country name";
+        if (!repository.containsCountry(countryId)) {
+            throw new IllegalArgumentException("The country is not found.");
```
Comment (gregorusanov) [inline src/main/java/com/ghilly/service/CountryServiceRest.java]: I don't think we need the exeption, it's an usual sitation you can retrun Null

File: src/main/java/com/ghilly/service/CountryServiceRest.java [hunk 4]
```diff
@@ -1,43 +1,58 @@
...
+        return repository.takeCountry(countryId);
     }
 
     @Override
-    public String upgrade(int countryId, String newCountryName) {
-        logger.info("The country with ID {} was upgraded, new name is {}.", countryId, newCountryName);
-        return newCountryName;
+    public void upgrade(int countryId, String newName) {
+        if (!repository.containsCountry(countryId)) {
+            throw new IllegalArgumentException("The country is not found.");
```
Comment (gregorusanov) [inline src/main/java/com/ghilly/service/CountryServiceRest.java]: pleae add countyId, without id we don't recognize what country is not found

File: src/main/java/com/ghilly/service/CountryServiceRest.java [hunk 5]
```diff
@@ -1,43 +1,58 @@
...
+        logger.info("The country with ID {} was upgraded, old name is {}, new name is {}.", countryId, oldName, newName);
     }
 
     @Override
-    public String remove(int countryId) {
-        logger.info("remove");
-        return "There will be the country name";
+    public void remove(int countryId) {
+        if (!repository.containsCountry(countryId)) {
+            throw new IllegalArgumentException("The country is not found.");
```
Comment (gregorusanov) [inline src/main/java/com/ghilly/service/CountryServiceRest.java]: pleae add countyId, without id we don't recognize what country is not found

File: src/test/java/com/ghilly/controller/CountryControllerTest.java [hunk 6]
```diff
@@ -12,70 +13,75 @@
...
         assertAll(
-                () -> verify(service).add(name),
+                () -> verify(service).add(NAME),
                 () -> verifyNoMoreInteractions(service)
         );
     }
 
     @Test
     void getCountries() {
         List<String> expected = new ArrayList<>(List.of("Japan", "Russia", "Germany"));
```
Comment (gregorusanov) [inline src/test/java/com/ghilly/controller/CountryControllerTest.java]: Why you use ArrayList, list.of return List

File: src/test/java/com/ghilly/service/CountryServiceRestTest.java [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+        assertAll(
+                () -> assertEquals(NAME, repository.takeCountry(ID)),
+                () -> verify(repository).insert(NAME)
+        );
+    }
+
+    @Test
+    void getAllCountries() {
+        List<String> expected = new ArrayList<>(List.of("Afghanistan", "France", "China"));
```
Comment (gregorusanov) [inline src/test/java/com/ghilly/service/CountryServiceRestTest.java]: Why you use ArrayList, list.of return List

**Review by gregorusanov — CHANGES_REQUESTED** [review]

File: src/main/java/com/ghilly/configuration/TravelDiaryConfiguration.java [hunk 8]
```diff
@@ -1,14 +1,21 @@
...
 public class TravelDiaryConfiguration {
 
+    @Bean
+    public CountryRepositoryRest repositoryRest() {
+        return new CountryRepositoryRest();
+    }
     @Bean
     public CountryServiceRest countryService() {
-        return new CountryServiceRest();
+        return new CountryServiceRest(repositoryRest());
```
Comment (gregorusanov) [inline src/main/java/com/ghilly/configuration/TravelDiaryConfiguration.java]: Don't use method in constuctor for Bean, please using the arguments...

File: src/main/java/com/ghilly/repository/CountryRepository.java (hunk 2 above)
Comment (ghilyana) [reply src/main/java/com/ghilly/repository/CountryRepository.java]: +

File: src/main/java/com/ghilly/service/CountryServiceRest.java (hunk 3 above)
Comment (ghilyana) [reply src/main/java/com/ghilly/service/CountryServiceRest.java]: +

File: src/main/java/com/ghilly/service/CountryServiceRest.java (hunk 4 above)
Comment (ghilyana) [reply src/main/java/com/ghilly/service/CountryServiceRest.java]: +

File: src/main/java/com/ghilly/service/CountryServiceRest.java (hunk 5 above)
Comment (ghilyana) [reply src/main/java/com/ghilly/service/CountryServiceRest.java]: +

File: src/test/java/com/ghilly/controller/CountryControllerTest.java (hunk 6 above)
Comment (ghilyana) [reply src/test/java/com/ghilly/controller/CountryControllerTest.java]: +

File: src/test/java/com/ghilly/service/CountryServiceRestTest.java (hunk 7 above)
Comment (ghilyana) [reply src/test/java/com/ghilly/service/CountryServiceRestTest.java]: +

**Review by gregorusanov — CHANGES_REQUESTED** [review]

**Review by gregorusanov — APPROVED** [review]

**Pull request merged**